client/my-sites/email/email-forwards-add/index.tsx

```diff
@@ -0,0 +1,121 @@
+import EmailForwardingAddNewCompactList from 'calypso/my-sites/email/email-forwarding/email-forwarding-add-new-compact-list';
+import EmailHeader from 'calypso/my-sites/email/email-header';
+import {
+	emailManagement,
+	emailManagementAddEmailForwards,
+	emailManagementPurchaseNewEmailAccount,
+} from 'calypso/my-sites/email/paths';
+import getCurrentRoute from 'calypso/state/selectors/get-current-route';
+import {
+	hasLoadedSiteDomains,
+	isRequestingSiteDomains,
+} from 'calypso/state/sites/domains/selectors';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+
+import './style.scss';
+
+type EmailForwardsAddProps = {
+	selectedDomainName: string;
+	source?: string;
+};
+
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+const noop = (): void => {};
+
+const EmailForwardsAdd = ( { selectedDomainName, source }: EmailForwardsAddProps ): JSX.Element => {
+	const currentRoute = useSelector( getCurrentRoute );
+	const selectedSite = useSelector( getSelectedSite );
+
+	const hasLoadedDomains = useSelector( ( state ) =>
+		hasLoadedSiteDomains( state, selectedSite?.ID ?? null )
+	);
+	const requestingSiteDomains = useSelector( ( state ) =>
+		Boolean( selectedSite && isRequestingSiteDomains( state, selectedSite.ID ) )
+	);
+	const areDomainsLoading = requestingSiteDomains || ! hasLoadedDomains;
+
+	const translate = useTranslate();
+
+	const goToEmail = useCallback( (): void => {
+		if ( ! selectedSite ) {
+			return;
+		}
+
+		if ( source === 'purchase' ) {
+			page(
+				emailManagementPurchaseNewEmailAccount(
+					selectedSite.slug,
+					selectedDomainName,
+					currentRoute
+				)
+			);
+			return;
+		}
+
+		page( emailManagement( selectedSite.slug, selectedDomainName, currentRoute ) );
+	}, [ currentRoute, selectedDomainName, selectedSite, source ] );
+
+	const onAddEmailForwardSuccess = useCallback( () => {
+		if ( ! selectedSite ) {
+			return;
+		}
+
+		page( emailManagement( selectedSite.slug, selectedDomainName, currentRoute ) );
+	}, [ currentRoute, selectedDomainName, selectedSite ] );
+
+	const analyticsPath = emailManagementAddEmailForwards( ':site', ':domain', currentRoute );
+
+	return (
+		<div className="email-forwards-add">
+			<PageViewTracker path={ analyticsPath } title="Email Management > Add Email Forwards" />
+
+			<QueryProductsList />
+
+			{ selectedSite && <QuerySiteDomains siteId={ selectedSite.ID } /> }
+
+			<Main wideLayout={ true }>
+				<DocumentHead title={ translate( 'Add New Email Forwards' ) } />
+
+				<EmailHeader currentRoute={ currentRoute } selectedSite={ selectedSite } />
+
+				<HeaderCake onClick={ goToEmail }>
+					{ translate( 'Email Forwarding' ) + ': ' + selectedDomainName }
+				</HeaderCake>
+
+				<SectionHeader label={ translate( 'Add New Email Forwards' ) } />
+
+				<Card>
+					{ areDomainsLoading && (
+						<div className="email-forwards-add__placeholder">
+							<p>&nbsp;</p>
+							<p>&nbsp;</p>
+							<Button disabled>&nbsp;</Button>
```

Comment:
Why do we need the literal space entities? 
```suggestion
							<p />
							<p />
							<Button disabled />
```

Reply:
Turns out we don't! I added them to force some content to start with, but that doesn't seem necessary. I've removed all three `&nbsp;` characters.